### PR TITLE
Running perf monitor on an independent node

### PR DESCRIPTION
**Pull request by Stephen1993**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/workflows/sdk-cli-perf-monitor-test.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# execute tests in src/promptflow/tests/sdk_cli_azure_test
```
Comment (wangchao1230) [inline .github/workflows/sdk-cli-perf-monitor-test.yml:1]: update this

Comment (Stephen1993) [reply .github/workflows/sdk-cli-perf-monitor-test.yml]: done.

**Review by wangchao1230 — APPROVED** [review]

File: .github/workflows/sdk-cli-perf-monitor-test.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        with:
+          pythonVersion: ${{ steps.display_env.outputs.pyVersion }}
+
+      - name: Build wheel
+        uses: "./.github/actions/step_sdk_setup"
+        with:
+          setupType: ${{ env.packageSetupType }}
+          scriptPath: ${{ env.testWorkingDirectory }}
+
+      - name: Get number of CPU cores
```
Comment (crazygao) [inline .github/workflows/sdk-cli-perf-monitor-test.yml:55]: remove this

File: .github/workflows/sdk-cli-perf-monitor-test.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        working-directory: ${{ env.testWorkingDirectory }}
+        run: |
+          gci env:* | sort-object name
+          python "../../scripts/building/run_coverage_tests.py" `
+            -p promptflow `
+            -t ${{ github.workspace }}/src/promptflow/tests/sdk_cli_azure_test ${{ github.workspace }}/src/promptflow/tests/sdk_cli_test `
+            -l eastus `
+            -m "perf_monitor_test" `
+            -n ${{ steps.cpu-cores.outputs.count }} `
+            --coverage-config ${{ github.workspace }}/src/promptflow/tests/sdk_cli_test/.coveragerc
```
Comment (crazygao) [inline .github/workflows/sdk-cli-perf-monitor-test.yml:80]: remove from -n

File: .github/workflows/sdk-cli-perf-monitor-test.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# execute tests in src/promptflow/tests/sdk_cli_azure_test, src/promptflow/tests/sdk_cli_test with mark perf_monitor_test
+# when pull_request triggers, use live mode
+
+name: sdk-cli-perf-monitor-test
+
+
+on:
+  pull_request:
+    branches:
```
Comment (crazygao) [inline .github/workflows/sdk-cli-perf-monitor-test.yml:9]: change to path filter

File: .github/workflows/sdk-cli-perf-monitor-test.yml (hunk 3 above)
Comment (Stephen1993) [reply .github/workflows/sdk-cli-perf-monitor-test.yml]: Is this to filter out directories that do not need to be searched or directories that need to be searched when searching for tests?

File: .github/workflows/sdk-cli-perf-monitor-test.yml (hunk 4 above)
Comment (Stephen1993) [reply .github/workflows/sdk-cli-perf-monitor-test.yml]: ok, done.

File: .github/workflows/sdk-cli-perf-monitor-test.yml (hunk 2 above)
Comment (Stephen1993) [reply .github/workflows/sdk-cli-perf-monitor-test.yml]: done.

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**